src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq.c

```diff
@@ -2092,6 +2200,116 @@ static ssize_t vmr_debug_type_store(struct device *dev,
 }
 static DEVICE_ATTR_WO(vmr_debug_type);
 
+static ssize_t clk_scaling_stat_raw_show(struct device *dev,
```

Comment:
This is a VERY dangerous function for it could crash the system if the the buf is smaller then the strings being printed. 

In addition, it looks like it isn't being used.  Is this still needed?

Reply:
It is used to read all the default configurations of clock throttling feature, override limits etc using xgq's sysfs node "clk_scaling_stat_raw". Also, this sysfs node used in device_linux.cpp file to get all feature stats in single call.

bash-4.2# cat /sys/bus/pci/devices/0000:d8:00.0/xgq_vmr.m.54525952/clk_scaling_stat_raw
HAS_CLOCK_THROTTLING:1
CLOCK_THROTTLING_ENABLED:1
POWER_SHUTDOWN_LIMIT:69
TEMP_SHUTDOWN_LIMIT:100
POWER_THROTTLING_LIMIT:290
TEMP_THROTTLING_LIMIT:95
POWER_THROTTLING_OVRD_LIMIT:200
TEMP_THROTTLING_OVRD_LIMIT:80
POWER_THROTTLING_OVRD_ENABLE:200
TEMP_THROTTLING_OVRD_ENABLE:80
CLOCK_THROTTLING_MODE:2